include/fmt/format.h

```diff
@@ -3305,6 +3306,66 @@ arg_join<internal::iterator_t<const Range>, wchar_t> join(const Range& range,
   return join(std::begin(range), std::end(range), sep);
 }
 
+template <typename Char, typename... T> struct tuple_arg_join : internal::view {
+  const std::tuple<T...>& tuple;
+  basic_string_view<Char> sep;
+
+  tuple_arg_join(const std::tuple<T...>& t, basic_string_view<Char> s) : tuple{t}, sep{s} {}
+};
+
+template <typename Char, typename... T>
+struct formatter<tuple_arg_join<Char, T...>, Char>
+{
+  template <typename ParseContext>
+  FMT_CONSTEXPR auto parse(ParseContext& ctx) -> decltype(ctx.begin()) {
+    return ctx.begin();
+  }
+
+  template <typename FormatContext>
+  typename FormatContext::iterator format(const tuple_arg_join<Char, T...>& value,
+                                          FormatContext& ctx) {
+    return format(value, ctx, std::make_index_sequence<sizeof...(T)>{});
+  }
+
+  template <typename FormatContext, size_t... N>
+  typename FormatContext::iterator format(const tuple_arg_join<Char, T...>& value,
```

Comment:
This and `format_args` should be private.

Reply:
Good catch I'll make that change before checkin